add an option to plot network in other shapes

wangg12 opened the pull request:
from #663 
* modify the `plot_network`
* give an example in `example/notebooks/simple_bind.ipynb`

pluskid commented on python/mxnet/visualization.py:
```diff
@@ -59,8 +64,17 @@ def plot_network(symbol, title="plot", shape=None):
+                node_attr[k] = node_attrs[k]
+            else:
+                raise ValueError("Input keys should be in: \
+                shape, fixedsize, width, height, style")
```
I guess it might be OK to just use `node_attr.update(node_attrs)` to merge whatever the user passed here, this allows the user to add extra parameters to GraphViz if needed.

wangg12 replied:
@pluskid  Got it.

pluskid commented:
looks good to me.

wangg12 commented:
@pluskid I don't know why it failed.

mli commented on mshadow:
```diff
@@ -1 +1 @@
-Subproject commit bc5cb993627f7a6633251f752ac93b0157e9f085
+Subproject commit 74be312ab6f20178766901a7caf021d4829e9110
```
this is reason why it is failed. to fix it, you changed mshadow to an older version. to fix it

```
cd mshadow
git pull
```

and then merge the changes of mshadow (which is a pointer to a commit)

wangg12 replied:
@mli Thx!

antinucleon commented:
This is good to be merged, can you do a rebase to squash the commits to one commit message?
http://mxnet.readthedocs.org/en/latest/contribute.html#how-to-combine-multiple-commits-into-one

Thanks!

wangg12 commented:
@antinucleon  I am still not sure how to squash these commits into one. I did all my pull requests online.

tqchen commented:
@wangg12  The instructions are here http://mxnet.readthedocs.org/en/latest/contribute.html#how-to-combine-multiple-commits-into-one

wangg12 commented:
I use `git rebase -i HEAD~x`(I don't know which x is right, so I tried 16). Then I found my first commit "add option to plot network in other shapes", deleted those commits before it and changed those after it to `squash`. But then I got error: `error: could not apply e83622c... overload other operators`. What should I do correctly?

tqchen commented:
There is conflicts with the master, that you might need to solve. Sorry for all the troubles

http://mxnet.readthedocs.org/en/latest/contribute.html#how-to-resolve-conflict-with-master

tqchen commented:
Maybe easier way might be keep what you have and start from a cleaner fork, if the rebase does not work out

wangg12 commented:
@tqchen How to start from a clean fork. Do I need to delete my former fork?

tqchen commented:
That is one way I can think of. Normally I use the rebase workflow to pull changes from upstream instead of PR back, which is easier to squash changes

wangg12 commented:
@tqchen  Thanks. I create a new pull request from a clean fork just now.
